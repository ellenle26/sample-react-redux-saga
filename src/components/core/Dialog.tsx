import classNames from "classnames";
import { FC } from "react";

import Button from "./Button";

const Dialog: FC<DialogProps> = (props: DialogProps) => {


    return (
        <div id="dialog" className={classNames({"hidden": !props.opened}, "absolute top-14 left-0 w-full h-[calc(100%-56px)] bg-black_70")}>
            <div className={classNames({"w-fit": props.width == null}, `${props.width}`, "max-w-4xl p-6 border-white rounded-xl relative top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 bg-white max-h-dialog overflow-auto")}>
                <div className="font-bold text-primary mb-5">{props.title}</div>
                <div>{props.children}</div>
                <div className="w-full flex justify-center mt-10">
                    <Button text={props.cancelText} type="outline" onClick={() => props.onClose && props.onClose()} 
                        className={classNames({ "mr-20": props.functionBttnMargin == null }, "font-bold text-primary border-primary rounded-button px-6 py-4 max-h-14")}/>
                    <Button text={props.saveText} type="contained" onClick={() => props.onSave && props.onSave()}
                        className="font-bold text-white bg-primary border-primary rounded-button px-6 py-4 max-h-14"/>
                </div>
            </div>
        </div>
    )
}

interface DialogProps {
    title?: string,
    children: any;
    opened: boolean,
    width?: string,
    saveText: string,
    cancelText: string,
    functionBttnMargin?: string,
    onSave?: () => void,
    onClose?: () => void
}

export default Dialog;