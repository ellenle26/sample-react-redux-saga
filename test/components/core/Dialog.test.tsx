/**
 * @jest-environment jsdom
*/


import Dialog from "@components/core/Dialog";
import { fireEvent, render } from "@testing-library/react";
import React from "react";

test("snapshot testing", () => {
    const renderResult = render(<Dialog opened={true} cancelText="Cancel" saveText="Save">Dialog content</Dialog>);
    expect(renderResult.asFragment()).toMatchSnapshot();
})

test("onCancel testing", () => {
    const handleCancel = jest.fn();
    const renderResult = render(<Dialog opened={true} cancelText="Cancel" saveText="Save" onClose={handleCancel}>Dialog content</Dialog>);
    const cancelBttn = renderResult.container.getElementsByTagName("button")[0];
    fireEvent.click(cancelBttn);
    expect(cancelBttn.innerHTML).toMatch("Cancel");
    expect(handleCancel).toHaveBeenCalledTimes(1);
})

test("onSave testing", () => {
    const handleSave = jest.fn();
    const renderResult = render(<Dialog opened={true} cancelText="Cancel" saveText="Save" onSave={handleSave
}>Dialog content</Dialog>);
    const saveBttn = renderResult.container.getElementsByTagName("button")[1];
    fireEvent.click(saveBttn);
    expect(saveBttn.innerHTML).toMatch("Save");
    expect(handleSave).toHaveBeenCalledTimes(1);
})