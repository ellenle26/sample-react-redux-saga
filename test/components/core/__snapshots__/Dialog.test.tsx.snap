// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing 1`] = `
<DocumentFragment>
  <div
    class="absolute top-14 left-0 w-full h-[calc(100%-56px)] bg-black_70"
    id="dialog"
  >
    <div
      class="w-fit undefined max-w-4xl p-6 border-white rounded-xl relative top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 bg-white max-h-dialog overflow-auto"
    >
      <div
        class="font-bold text-primary mb-5"
      />
      <div>
        Dialog content
      </div>
      <div
        class="w-full flex justify-center mt-10"
      >
        <button
          class="mr-20 font-bold text-primary border-primary rounded-button px-6 py-4 max-h-14 py-5 px-2 w-40 text-sm font-normal border border-solid hover:opacity-80"
        >
          Cancel
        </button>
        <button
          class="font-bold text-white bg-primary border-primary rounded-button px-6 py-4 max-h-14 py-5 px-2 w-40 text-sm font-normal border border-solid hover:opacity-80 bg-gray_"
        >
          Save
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
